[Bug]: GroupBy.quantile() fails for pandas implementation

### Describe the bug

The quantile groupby aggregation fails for the pandas backend:
```
df.group_by("a").agg(nw.col("b").quantile(quantile=0.3, interpolation="linear"))
```
```      
            select = names[0] if len(names) == 1 else list(names)
>           result = self.native_agg()(grouped[select])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           TypeError: GroupBy.quantile() got an unexpected keyword argument 'quantile'

narwhals/_pandas_like/group_by.py:161: TypeError
```

The issue is that pandas' quantile function expects the quantile with the parameter `q`, not `quantile`. 
Working on a PR.

### Steps or code to reproduce the bug

```
df.group_by("a").agg(nw.col("b").quantile(quantile=0.3, interpolation="linear"))
```

### Expected results

A valid narwhals df with the quantiles per group.

### Actual results

```
            select = names[0] if len(names) == 1 else list(names)
>           result = self.native_agg()(grouped[select])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           TypeError: GroupBy.quantile() got an unexpected keyword argument 'quantile'

narwhals/_pandas_like/group_by.py:161: TypeError
```

### Please run narwhals.show_versions() and enter the output below.

```shell
System:
    python: 3.12.13 (main, Apr 14 2026, 14:29:00) [Clang 22.1.3 ]
executable: /home/pedvide/narwhals-dev/.venv/bin/python3
   machine: Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.39

Python dependencies:
     narwhals: 2.20.0
        numpy: 2.4.4
       pandas: 2.3.3
        modin: 0.37.1
         cudf:
      pyarrow: 24.0.0
      pyspark: 4.1.1
       polars: 1.40.1
         dask: 2026.3.0
       duckdb: 1.4.4
         ibis: 12.0.0
     sqlframe: 4.1.0
```

### Relevant log output

```shell

```

--- a/narwhals/_pandas_like/group_by.py
+++ b/narwhals/_pandas_like/group_by.py
@@ -66,30 +66,17 @@ _REMAP_ORDERED_INDEX: Mapping[NarwhalsAggregation, Literal[0, -1]] = {
 }
 
 
-def groupby_agg_kwargs_to_pandas_equivalent(
-    function_name: NativeAggregation, kwargs: ScalarKwargs
-) -> dict[str, Any]:
-    if function_name == "quantile":
-        assert "quantile" in kwargs  # noqa: S101
-        assert "interpolation" in kwargs  # noqa: S101
-        pandas_kwargs = {
-            "q": kwargs["quantile"],
-            "interpolation": kwargs["interpolation"],
-        }
-    else:  # sum, len, ...
-        pandas_kwargs = dict(kwargs)
-    return pandas_kwargs
-
-
 @lru_cache(maxsize=32)
 def _native_agg(name: NativeAggregation, /, **kwds: Unpack[ScalarKwargs]) -> _NativeAgg:
     if name == "nunique":
         return methodcaller(name, dropna=False)
+    if name == "quantile":
+        assert "quantile" in kwds  # noqa: S101
+        assert "interpolation" in kwds  # noqa: S101
+        return methodcaller(name, q=kwds["quantile"], interpolation=kwds["interpolation"])
     if not kwds or kwds.get("ddof") == 1:
         return methodcaller(name)
-
-    kwargs = groupby_agg_kwargs_to_pandas_equivalent(name, kwds)
-    return methodcaller(name, **kwargs)
+    return methodcaller(name, **kwds)
 
 
 class AggExpr:

--- a/narwhals/_pandas_like/group_by.py
+++ b/narwhals/_pandas_like/group_by.py
@@ -66,13 +66,30 @@ _REMAP_ORDERED_INDEX: Mapping[NarwhalsAggregation, Literal[0, -1]] = {
 }
 
 
+def groupby_agg_kwargs_to_pandas_equivalent(
+    function_name: NativeAggregation, kwargs: ScalarKwargs
+) -> dict[str, Any]:
+    if function_name == "quantile":
+        assert "quantile" in kwargs  # noqa: S101
+        assert "interpolation" in kwargs  # noqa: S101
+        pandas_kwargs = {
+            "q": kwargs["quantile"],
+            "interpolation": kwargs["interpolation"],
+        }
+    else:  # sum, len, ...
+        pandas_kwargs = dict(kwargs)
+    return pandas_kwargs
+
+
 @lru_cache(maxsize=32)
 def _native_agg(name: NativeAggregation, /, **kwds: Unpack[ScalarKwargs]) -> _NativeAgg:
     if name == "nunique":
         return methodcaller(name, dropna=False)
     if not kwds or kwds.get("ddof") == 1:
         return methodcaller(name)
-    return methodcaller(name, **kwds)
+
+    kwargs = groupby_agg_kwargs_to_pandas_equivalent(name, kwds)
+    return methodcaller(name, **kwargs)
 
 
 class AggExpr:

--- a/tests/expr_and_series/quantile_test.py
+++ b/tests/expr_and_series/quantile_test.py
@@ -58,9 +58,7 @@ def test_quantile_expr(
 def test_quantile_expr_group_by(
     constructor: Constructor, request: pytest.FixtureRequest
 ) -> None:
-    if any(
-        x in str(constructor) for x in ("dask", "duckdb", "ibis", "pyspark", "pyarrow")
-    ):
+    if any(x in str(constructor) for x in ("dask", "pyspark", "pyarrow_table")):
         request.applymarker(pytest.mark.xfail)
 
     expected = {"a": [1, 2, 3], "b": [4.0, 6.0, 4.0]}

--- a/tests/expr_and_series/quantile_test.py
+++ b/tests/expr_and_series/quantile_test.py
@@ -58,7 +58,7 @@ def test_quantile_expr(
 def test_quantile_expr_group_by(
     constructor: Constructor, request: pytest.FixtureRequest
 ) -> None:
-    if any(x in str(constructor) for x in ("dask", "pyspark", "pyarrow_table")):
+    if any(x in str(constructor) for x in ("dask", "pyarrow_table")):
         request.applymarker(pytest.mark.xfail)
 
     expected = {"a": [1, 2, 3], "b": [4.0, 6.0, 4.0]}

--- a/tests/expr_and_series/quantile_test.py
+++ b/tests/expr_and_series/quantile_test.py
@@ -55,6 +55,26 @@ def test_quantile_expr(
         assert_equal_data(result, expected)
 
 
+def test_quantile_expr_group_by(
+    constructor: Constructor, request: pytest.FixtureRequest
+) -> None:
+    if any(
+        x in str(constructor) for x in ("dask", "duckdb", "ibis", "pyspark", "pyarrow")
+    ):
+        request.applymarker(pytest.mark.xfail)
+
+    expected = {"a": [1, 2, 3], "b": [4.0, 6.0, 4.0]}
+    q = 0.3
+    data = {"a": [1, 2, 3], "b": [4, 6, 4]}
+    df_raw = constructor(data)
+    df = nw.from_native(df_raw)
+
+    result = df.group_by("a").agg(
+        nw.col("b").quantile(quantile=q, interpolation="linear")
+    )
+    assert_equal_data(result.sort("a"), expected)
+
+
 @pytest.mark.parametrize(
     ("interpolation", "expected"),
     [

